src/Microsoft.DotNet.Build.Tasks.Feed/src/CreateAzureDevOpsFeed.cs

```diff
@@ -119,6 +120,14 @@ private async Task<bool> ExecuteAsync()
                 }
 
                 string azureDevOpsFeedsBaseUrl = $"https://feeds.dev.azure.com/{AzureDevOpsOrg}/";
+
+                Lazy<AzureCliCredential> credential = new();
```

Comment:
Maybe you wanted this?

```suggestion
                AzureCliCredential credential = new();
```

Reply:
I want to retain the option of using this MSBuild task in the old manner, where the AzureDevOpsPersonalAccessToken parameter is specified. By using Lazy<> the AzureCLI credentials are acquired only when necessary. I can still have a
```
AzureCliCredential credential = null;
if () {
credential = new AzureCliCredential();
}
```
but as for me, Lazy<> pattern looks better.

Reply:
I don't follow really.

What you have is functionally the same as this:

```
if (string.IsNullOrEmpty(AzureDevOpsPersonalAccessToken))
{
    const string AzureDevOpsScope = "499b84ac-1321-427f-aa17-267ca6975798/.default";
    AzureDevOpsPersonalAccessToken = new AzureCliCredential().GetToken(new TokenRequestContext(new[] { AzureDevOpsScope })).Token;
```

but deep down it:
- Creates the extra lazy object that is useless
- Needs to use reflection to create the AzureCliCredential as you don't provide the factory method

Reply:
Ok, agree.